model/internal/pdata/generated_metrics.go

```diff
@@ -1161,14 +1161,14 @@ func (ms NumberDataPoint) SetTimestamp(v Timestamp) {
 
 // ValueType returns the type of the value for this NumberDataPoint.
 // Calling this function on zero-initialized NumberDataPoint will cause a panic.
-func (ms NumberDataPoint) ValueType() MetricValueType {
+func (ms NumberDataPoint) ValueType() NumberType {
```

Comment:
What is the convention here? I want to make sure we follow a convention between `pdata.Value` and `pdata.Metric` for func and enum type names.

Reply:
Type accessor functions for `pdata.Metric`, `pdata.NumberDataPoint` and `pdata.Examplar` (except for `pdata.Value`) are called based on the oneof proto field name, but there is no convention for the type name.

Type accessor func naming:
- `pdata.Metric` has `DataType()`
- `pdata.NumberDataPoint` and `pdata.Exemplar` have `ValueType()`
- But `pdata.Value` has `Type()`

I don't think `pdata.Value` should follow the same convention for the type accessor. It would be `pdata.Value.ValueType()`, I think `Value` in the func name is redundant given that `pdata.Value` itself doesn't have any other fields as `pdata.Metric`, `pdata.NumberDataPoint` and `pdata.Exemplar` have.

Enum type naming:
There is no convention for that. We just use names that makes sense.
- `pdata.Metric.DataType()` returns `pdata.MetricDataType`
- `pdata.NumberDataPoint.ValueType()` and `pdata.Exemplar.ValueType()` return `pdata.NumberType`
- `pdata.Value.Type()` returns `ValueType`
Probably in order to adopt any convention here, we would need to split `pdata.NumberType` and use different types for `pdata.NumberDataPoint.ValueType()` and `pdata.Exemplar.ValueType()`.

Please let me know WDYT.

Reply:
Also I think the `NumberType` definitions should be moved to pdata/common

Reply:
@bogdandrutu please let me know WDYT ^